### DISCUSSION TITLE
KeyDown does not work anymore

### DISCUSSION THREAD

**Discussion by NPYisKewl** [bonsai-rx/bonsai#1175] [Q&A]
hello, my workflow (attached, first photo) has a key down node.  After the update, the node is now red and won't work(attached, first photo).  I did try similar nodes in the Toolbox to replace it (second photo) but none worked.  Any recommendations on fixing this? my workflow is in a zip. 
Thank you ! 

PJP

![Capture](https://user-images.githubusercontent.com/119971307/206006640-7ef66d91-3b30-4b49-8754-f6ef6380eb09.PNG)

![IMG_7150](https://user-images.githubusercontent.com/119971307/206008026-4f977b7a-376b-47e5-a533-1db38c009351.JPG)

[Airpuff_12062022_PJP.bonsai.zip](https://github.com/bonsai-rx/bonsai/files/10169798/Airpuff_12062022_PJP.bonsai.zip)

**Comment by bruno-f-cruz** [comment]
Heya
It looks like you are missing a package. I would just delete bonsai and reinstall the latest version. Make sure you install the "StarterPackage", it should have everything you need.
Also, the shader nodes you are trying to use are meant to work with a display window created by `CreateWindow`. Without it, they wont emit outputs. [Check the Bonvision website for examples.](https://bonvision.github.io/)

**Reply by NPYisKewl** [reply]
thank you for your help. I uninstalled Bonsai, and went into the C: pathway (photo attached) and searched for "Bonsai," "Neurophotometrics," and "nuget," and deleted the files. I then did a fresh install on Bonsai from their website. I tried to search for that package you were talking about but i get, what looks to me, a path error? Any recs? Much obliged. 

-grad student


![image](https://user-images.githubusercontent.com/119971307/206504266-e15db4c7-b9d7-4903-beca-2be1699fc9af.png)
